sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/FileMetaCacheReadBenchmark.scala

```diff
@@ -0,0 +1,129 @@
+
+    // Set default configs. Individual cases will change them if necessary.
+    sparkSession.conf.set(SQLConf.ORC_FILTER_PUSHDOWN_ENABLED.key, "true")
+    sparkSession.conf.set(SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key, "true")
+    sparkSession.conf.set(SQLConf.FILE_META_CACHE_TTL_SINCE_LAST_ACCESS.key, "5")
+
+    sparkSession
+  }
+
+  def withTempTable(tableNames: String*)(f: => Unit): Unit = {
+    try f finally tableNames.foreach(spark.catalog.dropTempView)
+  }
+
+  private def prepareTable(dir: File, df: DataFrame, fileCount: Int): Unit =
+    saveAsOrcTable( df.repartition(fileCount).write, dir.getCanonicalPath + "/orc")
+
+  private def saveAsOrcTable(df: DataFrameWriter[Row], dir: String): Unit = {
+    df.mode("overwrite").option("compression", "snappy").orc(dir)
+    spark.read.orc(dir).createOrReplaceTempView("orcTable")
+  }
+
+  def fileScanBenchmark(values: Int, width: Int, fileCount: Int): Unit = {
+    val benchmark = new Benchmark(
+      s"Scan from $width columns with $fileCount files",
+      values,
+      output = output)
+
+    withTempPath { dir =>
+      withTempTable("t1", "orcTable") {
+        import spark.implicits._
+        val middle = width / 2
+        val selectExpr = (1 to width).map(i => s"value as c$i")
+        val dataFrame = spark.range(values).map(_ => Random.nextLong).toDF()
+        dataFrame.selectExpr(selectExpr: _*).createOrReplaceTempView("t1")
+
+        prepareTable(dir, spark.sql("SELECT * FROM t1"), fileCount)
+
+        val filter = {
+          val rows =
+            spark.sql(s"SELECT c1, count(*) FROM orcTable group by c1")
+              .collect()
+              .sortBy(r => r.getLong(1))
+          rows.head.getLong(0)
+        }
+
+        benchmark.addCase("ORC Vectorized Full Scan: fileMetaCacheEnabled = false") { _ =>
+          spark.sql(s"SELECT sum(c$middle) FROM orcTable").noop()
+        }
+
+        benchmark.addCase("ORC Vectorized Full Scan: fileMetaCacheEnabled = true") { _ =>
+          withSQLConf(SQLConf.FILE_META_CACHE_ENABLED.key -> "true") {
+            spark.sql(s"SELECT sum(c$middle) FROM orcTable").noop()
+          }
+        }
+
+        benchmark.addCase("ORC Vectorized Filter Scan: fileMetaCacheEnabled = false") { _ =>
+          spark.sql(s"SELECT sum(c$middle) FROM orcTable where c1 = $filter").noop()
+        }
+
+        benchmark.addCase("ORC Vectorized Filter Scan: fileMetaCacheEnabled = true") { _ =>
+          withSQLConf(SQLConf.FILE_META_CACHE_ENABLED.key -> "true") {
+            spark.sql(s"SELECT sum(c$middle) FROM orcTable where c1 = $filter").noop()
+          }
+        }
+
+        benchmark.run()
+      }
+    }
+  }
+
+  override def runBenchmarkSuite(mainArgs: Array[String]): Unit = {
+    for (fileCount <- List(100, 500, 1000)) {
+      runBenchmark(s"Scan From $fileCount files") {
+        for (columnWidth <- List(10, 50, 100)) {
+          fileScanBenchmark(1024 * 1024 * 5, columnWidth, fileCount)
```

Comment:
Could you add a simple `countBenchmark`, @LuciferYang ?
I guess your PR's contribution will improve the performance of `count(*)` queries without `filter`s most significantly when we need only read footer information.

Reply:
2b99983 update to use a simple `countBenchmark`

Reply:
Interestingly, I found that the data from the server is not as good as Apple M1, haha